src/main/java/com/opencart/steps/RegisterPageBL.java

```diff
@@ -16,105 +16,36 @@ public RegisterPageBL() {
-        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
-        inputLastName(registerModel.getLastName());
-        inputEmail(registerModel.getEmail());
-        inputTelephone(registerModel.getTelephone());
-        inputPasswordAndConfirmPassword(registerModel.getPassword());
-        clickPolicyCheckbox();
-        clickOnContinueButton();
-        return this;
-
-
-    }
-
-    public RegisterPageBL registerNewPersonWithoutLastName() {
-        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
-        inputFirstName(registerModel.getFirstName());
-        inputEmail(registerModel.getEmail());
-        inputTelephone(registerModel.getTelephone());
-        inputPasswordAndConfirmPassword(registerModel.getPassword());
-        clickPolicyCheckbox();
-        clickOnContinueButton();
-        return this;
-    }
-
-    public RegisterPageBL registerNewPersonWithoutEmail() {
-        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
-        inputFirstName(registerModel.getFirstName());
-        inputLastName(registerModel.getLastName());
-        inputTelephone(registerModel.getTelephone());
-        inputPasswordAndConfirmPassword(registerModel.getPassword());
-        clickPolicyCheckbox();
-        clickOnContinueButton();
-        return this;
-    }
-
-    public RegisterPageBL registerNewPersonWithoutTelephone() {
-        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
-        inputFirstName(registerModel.getFirstName());
-        inputLastName(registerModel.getLastName());
-        inputEmail(registerModel.getEmail());
-        inputPasswordAndConfirmPassword(registerModel.getPassword());
-        clickPolicyCheckbox();
-        clickOnContinueButton();
-        return this;
-    }
 
-    public RegisterPageBL registerNewPersonWithoutPassword() {
-        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
-        inputFirstName(registerModel.getFirstName());
-        inputLastName(registerModel.getLastName());
-        inputEmail(registerModel.getEmail());
-        inputTelephone(registerModel.getTelephone());
-        clickPolicyCheckbox();
-        clickOnContinueButton();
         return this;
     }
 
-    public RegisterPageBL registerNewPersonWithoutVerifyPassword() {
-        RegisterModel registerModel = RegisterModelRepository.getRegisterModel();
+    public RegisterPageBL userRegisterWithoutPrivacy(RegisterModel registerModel) {
         inputFirstName(registerModel.getFirstName());
         inputLastName(registerModel.getLastName());
         inputEmail(registerModel.getEmail());
         inputTelephone(registerModel.getTelephone());
         inputPassword(registerModel.getPassword());
-        clickPolicyCheckbox();
+        inputConfirmPassword(registerModel.getPasswordConfirm());
         clickOnContinueButton();
+
         return this;
     }
```

Comment:
Please, make clickPolicyCheckbox() public, and click on it from tests when it needs, instead of creating this method. So, in the test where you call this method, you just won't click on PolicyCheckbox